portal/views/portal.py

```diff
@@ -268,20 +268,24 @@ def verify_token(valid_seconds):
             session['invited_verified_user_id'] = user.id
             return redirect(url_for('user.register', email=user.email))
 
-        # fall into else case below, at least for todays demo
-        # TODO: determine if there's a legit reason to login WRITE_ONLY
-        # w/o redirecting to challenge in this case
-
-    # Without WRITE_ONLY, we don't log the user in, but preserve the
-    # invited user id, should we need to merge associated details
-    # after user proves themselves and logs in
-    auditable_event("invited user entered using token, pending "
-                    "registration", user_id=user.id, subject_id=user.id,
-                    context='account')
-    session['challenge.user_id'] = user.id
-    session['challenge.next_url'] = url_for('user.register', email=user.email)
-    session['challenge.merging_accounts'] = True
-    return redirect(url_for('portal.challenge_identity'))
+        # If WRITE_ONLY user does not have PROMOTE_WITHOUT_IDENTITY_CHALLENGE,
+        # challenge the user identity, followed by a redirect to the
+        # registration page. Preserve the invited user id, should we need to
+        # merge associated details after user proves themselves and logs in
+        auditable_event("invited user entered using token, pending "
+                        "registration", user_id=user.id, subject_id=user.id,
+                        context='account')
+        existing_user = User.query.filter_by(email=user.email).first()
```

Comment:
@pbugni Ah yea, whoops, that line was leftover from an earlier attempt at this logic (it's not even used or referenced anywhere at this point). Should be removed now.